packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -0,0 +1,358 @@
+      case Axis.vertical:
+        return _computeIntrinsicWidthForHeight(height);
+    }
+    return null;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    switch (_direction) {
+      case Axis.horizontal:
+        return _childIterator
+                 .map/*<double>*/((RenderBox child) => child.getMaxIntrinsicWidth(double.INFINITY))
+                 .reduce((double value, double element) => value + element);
+      case Axis.vertical:
+        return _computeIntrinsicWidthForHeight(height);
+    }
+    return null;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    switch (_direction) {
+      case Axis.horizontal:
+        return _computeIntrinsicHeightForWidth(width);
+      case Axis.vertical:
+        return _childIterator
+                 .map/*<double>*/((RenderBox child) => child.getMinIntrinsicHeight(double.INFINITY))
+                 .reduce((double value, double element) => math.max(value, element));
+    }
+    return null;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    switch (_direction) {
+      case Axis.horizontal:
+        return _computeIntrinsicHeightForWidth(width);
+      case Axis.vertical:
+        return _childIterator
+                 .map/*<double>*/((RenderBox child) => child.getMaxIntrinsicHeight(double.INFINITY))
+                 .reduce((double value, double element) => value + element);
+    }
+    return null;
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return defaultComputeDistanceToHighestActualBaseline(baseline);
+  }
+
+  double _getCrossSize(RenderBox child) {
+    switch (_direction) {
+      case Axis.horizontal:
+        return child.size.height;
+      case Axis.vertical:
+        return child.size.width;
+    }
+    return null;
+  }
+
+  double _getMainSize(RenderBox child) {
+    switch (_direction) {
+      case Axis.horizontal:
+        return child.size.width;
+      case Axis.vertical:
+        return child.size.height;
+    }
+    return null;
+  }
+
+  @override
+  void performLayout() {
+    BoxConstraints childConstraints;
+    double maxExtent;
+    // TODO(ianh): Make the asserts below explain that there's no paint putting a Wrap in an unconstrained main axis (since then it's just a poor man's Flex).
+    switch (_direction) {
+      case Axis.horizontal:
+        assert(constraints.maxWidth.isFinite);
+        childConstraints = new BoxConstraints(maxWidth: constraints.maxWidth);
+        maxExtent = constraints.maxWidth;
+        break;
+      case Axis.vertical:
+        assert(constraints.maxHeight.isFinite);
+        childConstraints = new BoxConstraints(maxHeight: constraints.maxHeight);
+        maxExtent = constraints.maxHeight;
+        break;
+    }
+    assert(childConstraints != null);
+    assert(maxExtent != null);
+    RenderBox child = firstChild;
+    List<RenderBox> row = <RenderBox>[];
```

Comment:
You might call this a `run` rather than a `row` given that it can go vertically.